app/javascript/controllers/evaluation_criteria_controller.js

```diff
@@ -0,0 +1,221 @@
+    accordionContent.setAttribute("id", accordionId);
+
+    newCriteria.querySelectorAll("[id]").forEach((el) => {
+      el.id = el.id.replace("NEW_CRITERIA", this.counter);
+    });
+
+    newCriteria.querySelectorAll("[name]").forEach((el) => {
+      el.name = el.name.replace("NEW_CRITERIA", this.counter);
+    });
+
+    newCriteria.querySelectorAll("label").forEach((label) => {
+      label.setAttribute(
+        "for",
+        label.getAttribute("for").replace("NEW_CRITERIA", this.counter)
+      );
+    });
+  }
+
+  updateCriteriaTitles() {
+    this.visibleRows().forEach((row, index) => {
+      const numberElement = row.querySelector(".criteria-number");
+      numberElement.textContent = index + 1;
+    });
+  }
+
+  updateScoringOptions(row, scoringType) {
+    const options = {
+      scaleOptions: row.querySelector(".criteria-scale-options"),
+      binaryOptions: row.querySelector(".criteria-binary-options"),
+      ratingOptions: row.querySelector(".criteria-rating-options"),
+      scaleOptionLabels: row.querySelector(".criteria-scale-option-labels"),
+    };
+
+    console.log(options);
+
+    switch (scoringType) {
+      case "binary":
+        this.showBinaryOptions(options);
+        this.toggleOptionLabels(row, 0, 1);
+        break;
+      case "rating":
+        this.showRatingOptions(row, options);
+        break;
+      default:
+        this.hideAllOptions(options);
+        break;
+    }
+  }
+
+  showBinaryOptions(options) {
+    options.scaleOptions.style.display = "block";
+    options.binaryOptions.style.display = "block";
+    options.ratingOptions.style.display = "none";
+    this.enableInputs(options.binaryOptions);
+    this.disableInputs(options.ratingOptions);
+  }
+
+  showRatingOptions(row, options) {
+    options.scaleOptions.style.display = "block";
+    options.binaryOptions.style.display = "none";
+    options.ratingOptions.style.display = "block";
+    this.enableInputs(options.ratingOptions);
+    this.disableInputs(options.binaryOptions);
+    const start = parseInt(
+      row.querySelector(".option-range-select.option-range-start").value
+    );
+    const end = parseInt(
+      row.querySelector(".option-range-select.option-range-end").value
+    );
+    this.toggleOptionLabels(row, start, end);
+  }
+
+  hideAllOptions(options) {
+    options.scaleOptions.style.display = "none";
+    options.binaryOptions.style.display = "none";
+    options.ratingOptions.style.display = "none";
+    this.disableInputs(options.binaryOptions);
+    this.disableInputs(options.ratingOptions);
+    this.disableInputs(options.scaleOptionLabels);
+  }
+
+  toggleOptionLabels(row, start, end) {
+    row
+      .querySelectorAll(".criteria-option-label-row")
+      .forEach((labelRow, index) => {
+        labelRow.style.display =
+          index >= start && index <= end ? "flex" : "none";
+        const input = labelRow.querySelector("input");
+        input.disabled = !(index >= start && index <= end);
```

Comment:
this is fine, but for me it reads a little easier without negating the expression
```suggestion
        input.disabled = index < start || index > end;
```